client/src/components/SignUpForm.jsx

```diff
@@ -0,0 +1,58 @@
+import React, { useState } from "react";
+import axios from "axios";
+import { useNavigate } from "react-router-dom";
+
+const SignUpForm = ({ setUserExist }) => {
```

Comment:
```suggestion
const SignUpForm = ({ onUserExist }) => {
```

絶対ではないですが、Reactのスタンダードとしてイベントハンドラは `on` から始める事が多いです。